src/Entity/Tower.cpp

```diff
@@ -37,22 +37,35 @@ void PlaceNewTower(flecs::iter& iter)
 		return;
 
 	const Vector2 mousePosition = GetMousePosition();
-	const TransformComponent transformComponent{Vector2{mousePosition.x, mousePosition.y}, Vector2{TOWER_WIDTH * ENTITY_SCALE, TOWER_HEIGHT*ENTITY_SCALE}};
+	const TransformComponent transformComponent{Vector2{mousePosition.x, mousePosition.y}, Vector2{TOWER_WIDTH * ENTITY_SCALE, TOWER_HEIGHT * ENTITY_SCALE}};
+	
+	//QUESTION: I am using transform here, but should get the data from collision component. Considering it's always same, should I add global const parameters for it?
+	//what's the best way to "reuse it", considering i will in future have a collider that's different than sprite rect (a sphere on a slime)
+
 	const Rectangle newTowerRect{transformComponent.mPosition.x, transformComponent.mPosition.y, transformComponent.mScale.x, transformComponent.mScale.y};
 
 	bool hasCollided{false};
-
-	//Replaced collision component for now with transform component, same issue as Character.cpp checking on collision on update, cause cant pass two components in query
-	//Assuming at this point, that collision rect is the same as transform rect
-
-	iter.world().each<TransformComponent>([&hasCollided, &newTowerRect](TransformComponent& otherTransformComponent) {
-		// Avoid computing a collision detection if we already detected a collision.
+	Rectangle otherRect{};
+	auto filter = iter.world().filter<CollisionComponent, TransformComponent>();
+	
+	filter.iter([&hasCollided, &newTowerRect, &otherRect](flecs::iter& it, CollisionComponent* collisionComponents, TransformComponent* transformComponents) 
+	{
 		if (hasCollided)
 			return;
+		for (int i = 0; i < it.count(); ++i)
+		{
+		auto collisionComponent = collisionComponents[i];
+		auto transformComponent = transformComponents[i];
+
+		//QUESTION: how do i check that i am not checking the collision "with itself", aka i am checking Object
+		//with all other Objects, so, I am bound to get collision with "itself"?
 
-		const Rectangle otherRect = {otherTransformComponent.mPosition.x, otherTransformComponent.mPosition.y, otherTransformComponent.mScale.x, otherTransformComponent.mScale.y};
+		//QUESTION: I didnt want to create a variable every iteration in a for loop, afaik, creating it many times is worse than having non-const being passed?
```

Comment:
It's easy to assume this would be the case, but it's not what's happening. The variable will be allocated once if you have declared `Rectangle other` in or outside the loop.

The difference is that by having the declaration outside, the variable will be constructed/deconstructed once, but if it's inside the loop, it will be constructed/deconstructed for each loop.

So if your variable is a primitive or POD (plain old data, e.g. a `struct` of 4 `int`), it's fine to have the declaration within the `for` loop (it also helps to ensure you limit the scope, and the compiler could optimize stuff on its own as well).

If your variable is non-POD (e.g. a custom class), you might wanna do that optimization, need to consider what it does with its constructor/deconstructor/assignment operator.

For more info:
- https://stackoverflow.com/questions/7959573/declaring-variables-inside-loops-good-practice-or-bad-practice
- https://stackoverflow.com/questions/982963/is-there-any-overhead-to-declaring-a-variable-within-a-loop-c?noredirect=1&lq=1

You can also make use of https://godbolt.org to see what the compiler does. With a simple example, you can see the compiler has the same number of instructions whether it is declared inside or outside the loop: https://godbolt.org/z/zaKr5xPYs